responsibleai/responsibleai/_managers/causal_manager.py

```diff
@@ -2,30 +2,168 @@
 # Licensed under the MIT License.
 
 """Defines the Causal Manager class."""
+import numpy as np
+
+from econml.solutions.causal_analysis import CausalAnalysis
 
 from responsibleai._internal.constants import ManagerNames
 from responsibleai._managers.base_manager import BaseManager
+from responsibleai._config.base_config import BaseConfig
+from responsibleai.modelanalysis.constants import ModelTask
+from responsibleai.exceptions import UserConfigValidationException, DuplicateManagerConfigException
+
+
+class CausalConstants:
+    # Model types
+    AUTOML = 'automl'
+    LINEAR = 'linear'
+
+
+class CausalConfig(BaseConfig):
+    def __init__(
+        self,
+        features,
+        categorical_features,
+        treatment_features=None,
+        nuisance_model=None,
+        heterogeneity_model=None,
+    ):
+        super().__init__()
+        self.features = features
+        self.categorical_features = categorical_features
+        self.treatment_features = treatment_features
+        self.nuisance_model = nuisance_model
+        self.heterogeneity_model = heterogeneity_model
+
+        # Outputs
+        self.causal_effects = None
+        self.has_computation_failed = False
+        self.failure_reason = None
+        self.is_computed = False
+
+    def __eq__(self, other):
+        return (
+            self.features == other.features and
```

Comment:
what about the other parameters above, eg treatment_features, categorical_features, heterogeneity_model?

Reply:
Oh yes, added, thanks